core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala

```diff
@@ -32,7 +32,12 @@ private[ui] class JobPage(parent: JobsTab) extends WebUIPage("job") {
 
   def render(request: HttpServletRequest): Seq[Node] = {
     listener.synchronized {
-      val jobId = request.getParameter("id").toInt
+      val paramaterId = request.getParameter("id")
+      if (paramaterId.equals("")) {
```

Comment:
Yes, but don't you want the code to handle both cases? and test both cases?

On Wed, Jan 28, 2015 at 4:44 PM, Kirill A. Korinskiy <
notifications@github.com> wrote:

> In core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala
> <https://github.com/apache/spark/pull/4239#discussion_r23700631>:
>
> > @@ -32,7 +32,12 @@ private[ui] class JobPage(parent: JobsTab) extends WebUIPage("job") {
> >
> >    def render(request: HttpServletRequest): Seq[Node] = {
> >      listener.synchronized {
> > -      val jobId = request.getParameter("id").toInt
> > +      val paramaterId = request.getParameter("id")
> > +      if (paramaterId.equals("")) {
>
> @srowen <https://github.com/srowen>, yes, you are right.
>
> But for case &id= it's return "". And I used this case for testing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/4239/files#r23700631>.
>